packages/react-router/tests/link.test.tsx

```diff
@@ -1,44 +1,805 @@
     const indexRoute = createRoute({
       getParentRoute: () => rootRoute,
       path: '/',
       component: () => (
-        <Link to="/" disabled>
-          Index
-        </Link>
+        <React.Fragment>
+          <h1>Index</h1>
+          <Link to="/posts" disabled>
+            Posts
+          </Link>
+        </React.Fragment>
       ),
     })
+
+    const postsRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/posts',
+      component: () => (
+        <React.Fragment>
+          <h1>Posts</h1>
+        </React.Fragment>
+      ),
+    })
+
     const router = createRouter({
-      routeTree: rootRoute.addChildren([indexRoute]),
-      history: createMemoryHistory({ initialEntries: ['/'] }),
+      routeTree: rootRoute.addChildren([indexRoute, postsRoute]),
     })
 
-    await router.load()
+    render(<RouterProvider router={router} />)
 
-    const rendered = render(<RouterProvider router={router} />)
-    const customElement = rendered.queryByText('Index')
+    const postsLink = await screen.findByRole('link', { name: 'Posts' })
+
+    expect(window.location.pathname).toBe('/')
+
+    expect(postsLink).not.toBeDisabled()
+    expect(postsLink).toHaveAttribute('aria-disabled', 'true')
+
+    fireEvent.click(postsLink)
+
+    await expect(
+      screen.findByRole('header', { name: 'Posts' }),
+    ).rejects.toThrow()
+  })
+
+  it('when the current route is the root', async () => {
+    const rootRoute = createRootRoute()
+    const indexRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/',
+      component: () => {
+        return (
+          <React.Fragment>
+            <h1>Index</h1>
+            <Link to="/" activeProps={{ className: 'active' }}>
+              Index
+            </Link>
+            <Link to="/posts" inactiveProps={{ className: 'inactive' }}>
+              Posts
+            </Link>
+          </React.Fragment>
+        )
+      },
+    })
+
+    const postsRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/posts',
+      component: () => {
+        return (
+          <React.Fragment>
+            <h1>Posts</h1>
+          </React.Fragment>
+        )
+      },
+    })
+
+    const router = createRouter({
+      routeTree: rootRoute.addChildren([indexRoute, postsRoute]),
+    })
+
+    render(<RouterProvider router={router} />)
+
+    const indexLink = await screen.findByRole('link', { name: 'Index' })
+
+    expect(window.location.pathname).toBe('/')
+
+    expect(indexLink).toHaveAttribute('aria-current', 'page')
+    expect(indexLink).toHaveClass('active')
+    expect(indexLink).toHaveAttribute('data-status', 'active')
+    expect(indexLink).toHaveAttribute('href', '/')
 
-    expect(customElement!.hasAttribute('disabled')).toBe(false)
+    const postsLink = await screen.findByRole('link', { name: 'Posts' })
+
+    expect(postsLink).toHaveClass('inactive')
+    expect(postsLink).toHaveAttribute('href', '/posts')
```

Comment:
Could a test also be added in, testing the href value when a `basePath` value is set on the router.

```tsx
const router = createRouter({
  basePath: '/app'
})
```

Seems like a good runtime check to guarantee.